packages/gensx/src/commands/examples/clone.tsx

```diff
@@ -0,0 +1,131 @@
+import { exec } from "node:child_process";
+import fs from "node:fs";
+import path from "node:path";
+import { promisify } from "node:util";
+
+import { Box, Text, useApp } from "ink";
+import { useEffect, useState } from "react";
+
+import { ErrorMessage } from "../../components/ErrorMessage.js";
+import { LoadingSpinner } from "../../components/LoadingSpinner.js";
+import { getExampleByName } from "./supported-examples.js";
+
+const execAsync = promisify(exec);
+
+export interface CloneExampleOptions {
+  exampleName: string;
+  projectName?: string;
+  yes?: boolean;
+}
+
+export function CloneExampleUI({
+  exampleName,
+  projectName,
+  yes,
+}: CloneExampleOptions) {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+  const [success, setSuccess] = useState(false);
+  const [step, setStep] = useState("Validating example...");
+  const { exit } = useApp();
+
+  useEffect(() => {
+    let mounted = true;
+
+    async function cloneExample() {
+      try {
+        // Validate example exists and get example object
+        const example = getExampleByName(exampleName);
+
+        if (!example) {
+          throw new Error(
+            `Unknown example: ${exampleName}. Run 'gensx examples' to see available examples.`,
+          );
+        }
+
+        const targetDir = projectName ?? exampleName;
+        setStep("Checking target directory...");
+
+        // Check if target directory exists
+        const targetPath = path.resolve(targetDir);
+        if (fs.existsSync(targetPath)) {
+          throw new Error(
+            `Directory ${targetDir} already exists. Please choose a different name or remove the existing directory.`,
+          );
+        }
+
+        setStep("Cloning example...");
+
+        // Use degit to clone the example
+        const degitCommand = `npx degit ${example.path} ${targetDir}`;
```

Comment:
Does this assume they have cloned the gensx repo? The `path` in the supported examples collection is just a relative path, so how does this work for `npx gensx examples clone draft-pad`?

Reply:
NVM I misunderstood how degit works.